app/models/manageiq/providers/embedded_terraform/automation_manager/amazon_credential.rb

```diff
@@ -0,0 +1,68 @@
+class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::AmazonCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::CloudCredential
+  COMMON_ATTRIBUTES = [
+    {
+      :component  => 'text-field',
+      :label      => N_('Access Key'),
+      :helperText => N_('AWS Access Key for this credential'),
+      :name       => 'userid',
+      :id         => 'userid',
+      :isRequired => true,
+      :validate   => [{:type => 'required'}],
+    },
+    {
+      :component  => 'password-field',
+      :label      => N_('Secret Key'),
+      :helperText => N_('AWS Secret Key for this credential'),
+      :name       => 'password',
+      :id         => 'password',
+      :type       => 'password',
+      :isRequired => true,
+      :validate   => [{:type => 'required'}],
+    },
+  ].freeze
+
+  EXTRA_ATTRIBUTES = [
+    {
+      :component  => 'password-field',
+      :label      => N_('STS Token'),
+      :helperText => N_('Security Token Service(STS) Token for this credential'),
+      :name       => 'security_token',
+      :id         => 'security_token',
+      :type       => 'password',
+      :maxLength  => 1024
+    },
+    {
+      :component  => 'text-field',
+      :label      => N_('Region'),
+      :helperText => N_('AWS Region where the provider will operate. The Region must be set.'),
+      :name       => 'region',
+      :id         => 'region',
+      :isRequired => true,
+      :maxLength  => 50,
+    },
+  ].freeze
+
+  API_ATTRIBUTES = (COMMON_ATTRIBUTES + EXTRA_ATTRIBUTES).freeze
+
+  API_OPTIONS = {
+    :type       => 'cloud',
+    :label      => N_('Amazon'),
+    :attributes => API_ATTRIBUTES
+  }.freeze
+
+  alias security_token auth_key
+
+  def self.display_name(number = 1)
+    n_('Credential (Amazon)', 'Credentials (Amazon)', number)
+  end
+
+  def self.params_to_attributes(params)
+    attrs = super.dup
+    attrs[:auth_key] = attrs.delete(:security_token) if attrs.key?(:security_token)
```

Comment:
Why call a property security_token if we're just going to store it in auth_key, could you use `"auth_key"` instead up here :point_up: https://github.com/ManageIQ/manageiq-providers-embedded_terraform/pull/16/files#diff-10eac92f3aa8adb511c059784672e62020e8e6172d8154c295c6a5678c75007fR29-R30

Reply:
Good question. We also have an alias for security_token from auth_key so hopefully we can use auth_key everywhere to keep it consistent.

Reply:
Yeah I understand having the `:label` be something specific but wasn't sure what having the internal key being different from the column bought us

Reply:
Just followed the pattern of what is in ansible implemetation - just a copy the same 
https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/amazon_credential.rb#L52

Reply:
This can be done in a follow-up

Reply:
yup, I'll do this in a follow up. no need to delay merging on this.